module/data/user_settings.py

```diff
@@ -0,0 +1,129 @@
+            """)
+            return res.fetchall()
+
+    def getUsersToNofify(self, day: DAYS, meal: MEALS) -> list[str]:
+        if day not in VALID_DAYS:
+            raise ValueError("Parameter day must be a valid day")
+        if meal not in VALID_MEALS:
+            raise ValueError("Expected meal to be lunch or dinner")
+
+        day_meal = day + '_' + meal
+
+        with sqlite3.connect(DB_PATH) as con:
+            con.row_factory = self.row_factory
+            cur = con.cursor()
+            result = cur.execute("""SELECT chat_id
+            FROM user_settings
+            WHERE (?) = 1
+            """, (day_meal, ))
+            return result.fetchall()
+
+    def insert_user(self, chat_id:int) -> None:
+        with sqlite3.connect(DB_PATH) as con:
+            cur = con.cursor()
+            cur.execute("""SELECT chat_id
+            FROM user_settings
+            WHERE chat_id = (?)""", (chat_id, ))
+
+            user_exists = cur.fetchall()
+            if len(user_exists) == 0:
+                cur.execute("INSERT INTO user_settings (chat_id) values (?)", (chat_id, ))
+
+    def setMeal(self, chat_id: int, day: DAYS, meal: MEALS) -> None:
+        if day not in VALID_DAYS:
+            raise ValueError("Parameter day must be a valid day")
+        if meal not in VALID_MEALS:
+            raise ValueError("Expected meal to be lunch or dinner")
+
+        self.insert_user(chat_id)
+        day_meal = day + '_' + meal
+
+        with sqlite3.connect(DB_PATH) as con:
+            cur = con.cursor()
+            cur.execute(f"""SELECT {day_meal}
+            FROM user_settings
+            WHERE chat_id = {chat_id}
+            """)
+            old_meal = cur.fetchall()
+
+            if old_meal == [(0, )]:
+                cur.execute(f"""UPDATE user_settings
+                SET {day_meal} = 1
+                WHERE chat_id = {chat_id}
+                """)
+            if old_meal == [(1, )]:
+                cur.execute(f"""UPDATE user_settings
+                SET {day_meal} = 0
+                WHERE chat_id = {chat_id}
+                """)
+
+            cur.execute(f"""SELECT {day_meal}
+            FROM user_settings
+            WHERE chat_id = {chat_id}
+            """)
+            new_meal = cur.fetchall()
+            if new_meal == [(0, )]:
+                cur.execute("""DELETE FROM user_settings
+                WHERE (?) = chat_id
+                AND monday_lunch = 0
+                AND monday_dinner = 0
+
+                AND tuesday_lunch = 0
+                AND tuesday_dinner = 0
+
+                AND wednesday_lunch = 0
+                AND wednesday_dinner = 0
+
+                AND thursday_lunch = 0
+                AND thursday_dinner = 0
+
+                AND friday_lunch = 0
+                AND friday_dinner = 0
+
+                AND saturday_lunch = 0
+                AND saturday_dinner = 0
+
+                AND sunday_lunch = 0
+                AND sunday_dinner = 0
```

Comment:
not sure if we need this list of `AND`s

Reply:
I'm not sure too, I'll search a better way to do the same thing

Reply:
By the way, I removed it, the cancelation of a record may cause problems with the dynamic generation of the keyboard